File: 1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        unordered_map<int, int> mp;
        for (auto ch : nums) {
            mp[ch]++;
        }
        int res = 0;
        for (auto ch : nums) {
            int diff = k - ch;
            if (mp.find(diff) != mp.end() && mp[diff] > 0 && mp[ch] > 0 &&
                !(diff == ch && mp[diff] <= 1)) {
                res++;
                mp[diff]--;
                mp[ch]--;
            }
        }
        return res;
    }
};

File: 1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxOperations = function (nums, k) {
    const mp = {};
    let res = 0;
    nums.forEach((item) => {
      mp[item]=(mp[item] || 0)+1;
    })
    nums.forEach((item) => {
        const diff = k - item;
        if (mp[item] > 0 && mp[diff] > 0 && !(diff === item && mp[diff] <= 1)) {
            mp[item]--;
            mp[diff]--;
            res++;
        }
    })
    return res;

};